src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
+    const pendingCommentText = 'Example of a comment being edited';
+    const store = configureStore();
+    store.dispatch(
+      actions.beginSaveComment({ ...keyParams, pendingCommentText }),
+    );
+
+    const root = render({ readOnly: false, store, ...keyParams });
 
+    const textarea = root.find(`.${styles.textarea}`);
+    expect(textarea).toHaveProp('disabled', true);
+    expect(textarea).toHaveProp('value', pendingCommentText);
+
+    const button = root.find(Button);
+    expect(button).toHaveProp('disabled', true);
+    expect(button).toHaveText('Saving…');
+  });
+
+  it('seeds the form with an initial comment', () => {
+    const commentId = 1;
+    const commentText = 'Example of a previously saved comment';
+    const comment = createFakeExternalComment({
+      id: commentId,
+      comment: commentText,
+    });
+    const { store } = dispatchComment({ comment });
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const root = render({
+      _manageComment,
+      commentId,
+      store,
+    });
+
+    root.find(`.${styles.form}`).simulate('submit', createFakeEvent());
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_manageComment).toHaveBeenCalledWith(
+      expect.objectContaining({
+        comment: commentText,
+        commentId: comment.id,
+      }),
+    );
+  });
+
+  it('seeds the form with initialCommentText', () => {
+    const keyParams = { fileName: null, line: null, versionId: 1 };
+    const pendingCommentText = 'Example of a comment being edited';
+    const store = configureStore();
+    store.dispatch(
+      actions.beginSaveComment({ ...keyParams, pendingCommentText }),
+    );
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const root = render({
+      _manageComment,
+      store,
+      ...keyParams,
+    });
+
+    root.find(`.${styles.form}`).simulate('submit', createFakeEvent());
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_manageComment).toHaveBeenCalledWith(
+      expect.objectContaining({
+        comment: pendingCommentText,
+        commentId: undefined,
+      }),
+    );
+  });
+
+  it('lets you update a previously saved comment', () => {
+    const commentId = 1;
+    const store = configureStore();
+    const comment = createFakeExternalComment({
+      id: commentId,
+      comment: 'Example of a previously saved comment',
+    });
+    dispatchComment({ store, comment });
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const commentText = 'Example of an edited comment';
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const root = render({
+      _manageComment,
+      commentId,
+      store,
+    });
+
```

Comment:
Would it be worth having an assertion that the current contents of the `textarea` is the value from the existing comment? The earlier test for `seeds the form with an initial comment` doesn't check that either, so it seems like there's no coverage for that.

Reply:
Yeah, that will be helpful to prove that the textarea is part of the update process.